ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_op.hpp

```diff
@@ -6,89 +6,128 @@
-    const auto attribute_values() const {
-        return std::forward_as_tuple(
-            this->binary_op_type, this->in_place, this->activations, this->memory_config, this->dtype);
-    }
-};
-
 struct Binary {
-    const BinaryProgramConfig program_config;
-    std::optional<DeviceComputeKernelConfig> compute_kernel_config;
-
-    void validate_with_output_tensors(const std::vector<Tensor> &input_tensors, const std::vector<std::optional<Tensor>>& output_tensors) const;
-    std::vector<tt::tt_metal::Shape> compute_output_shapes(const std::vector<Tensor> &input_tensors) const;
-    std::vector<Tensor> create_output_tensors(const std::vector<Tensor> &input_tensors, const std::vector<std::optional<Tensor>>& output_tensors) const;
-    operation::ProgramWithCallbacks create_program(
-        const std::vector<Tensor> &input_tensors, std::vector<Tensor> &output_tensors) const;
-
-    const operation::Hash compute_program_hash(const std::vector<Tensor> &input_tensors) const;
-
-    operation::OpPerformanceModel create_op_performance_model(
-        const std::vector<Tensor> &input_tensors,
-        const std::vector<std::optional<const Tensor>> &optional_input_tensors,
-        std::vector<Tensor> &output_tensors) const;
-
-    static constexpr auto attribute_names = std::forward_as_tuple("program_config", "compute_kernel_config");
-    const auto attribute_values() const {
-        return std::forward_as_tuple(this->program_config, this->compute_kernel_config);
-    }
+    struct operation_attributes_t {
+        BinaryOpType binary_op_type;
+        bool in_place;
+        const std::optional<FusedActivations> activations;
+        const MemoryConfig memory_config;
+        const DataType dtype;
+        std::optional<DeviceComputeKernelConfig> compute_kernel_config;
+
+        static constexpr auto attribute_names = std::forward_as_tuple(
+            "binary_op_type", "in_place", "activations", "memory_config", "dtype", "compute_kernel_config");
+        const auto attribute_values() const {
+            return std::forward_as_tuple(
+                this->binary_op_type,
+                this->in_place,
+                this->activations,
+                this->memory_config,
+                this->dtype,
+                this->compute_kernel_config);
+        }
+    };
+    struct tensor_args_t {
+        const Tensor& input_tensor_a;
+        const Tensor& input_tensor_b;
+        std::optional<Tensor> output_tensor;
+
+        static constexpr auto attribute_names =
+            std::forward_as_tuple("input_tensor_a", "input_tensor_b", "output_tensor");
+        const auto attribute_values() const {
+            return std::forward_as_tuple(this->input_tensor_a, this->input_tensor_b, this->output_tensor);
+        }
+    };
+    using shape_return_value_t = ttnn::Shape;
+    using tensor_return_value_t = Tensor;
+
+    struct ElementWiseMultiCore {
```

Comment:
These structs are pretty dummy. Do they need to all be here? Can't go away to their correspondent program files?